app/scripts/controllers/metametrics-data-deletion/metametrics-data-deletion.ts

```diff
@@ -0,0 +1,220 @@
+import {
+  BaseController,
+  RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import type { DataDeletionService } from '../../services/data-deletion-service';
+
+// Unique name for the controller
+const controllerName = 'MetaMetricsDataDeletionController';
+
+/**
+ * Get a type representing the public interface of the given type. The
+ * returned type will have all public properties, but will omit private
+ * properties.
+ *
+ * @template Interface - The interface to return a public representation of.
+ */
+type PublicInterface<Interface> = Pick<Interface, keyof Interface>;
+
+/**
+ * @type DataDeleteDate
+ * Timestamp at which regulation response is returned.
+ */
+export type DataDeleteDate = number;
+/**
+ * @type DataDeleteDate
+ * Regulation Id retuned while creating a delete regulation.
+ */
+export type DataDeleteRegulationId = string;
+
+/**
+ * The status on which to filter the returned regulations.
+ * Mentioned here: https://docs.segmentapis.com/tag/Deletion-and-Suppression#operation/listRegulationsFromSource
+ *
+ * @type DeleteRegulationStatus
+ */
+export enum DeleteRegulationStatus {
+  FAILED = 'FAILED',
+  FINISHED = 'FINISHED',
+  INITIALIZED = 'INITIALIZED',
+  INVALID = 'INVALID',
+  NOT_SUPPORTED = 'NOT_SUPPORTED',
+  PARTIAL_SUCCESS = 'PARTIAL_SUCCESS',
+  RUNNING = 'RUNNING',
+  UNKNOWN = 'UNKNOWN',
```

Comment:
nit: I believe we generally use PascalCase for enum keys.
(e.g. `DeleteRegulationStatus.PartialSuccess` not `DeleteRegulationStatus.PARTIAL_SUCCESS`)

```suggestion
  Failed = 'FAILED',
  Finished = 'FINISHED',
  Initialized = 'INITIALIZED',
  Invalid = 'INVALID',
  NotSupported = 'NOT_SUPPORTED',
  PartialSuccess = 'PARTIAL_SUCCESS',
  Running = 'RUNNING',
  Unknown = 'UNKNOWN',
```